debsources/app/patches/routes.py

```diff
@@ -0,0 +1,112 @@
+# redistribute it and/or modify it under the terms of the GNU Affero General
+# Public License as published by the Free Software Foundation, either version 3
+# of the License, or (at your option) any later version.  For more information
+# see the COPYING file at the top-level directory of this distribution and at
+# https://anonscm.debian.org/gitweb/?p=qa/debsources.git;a=blob;f=COPYING;hb=HEAD
+
+
+from __future__ import absolute_import
+
+from flask import jsonify
+
+from . import bp_patches
+
+from ..helper import bind_render
+from ..views import (IndexView, Ping, PrefixView, ErrorHandler,
+                     ListPackagesView, PackageVersionsView)
+from .views import SummaryView
+
+
+# context vars
+@bp_patches.context_processor
+def skeleton_variables():
+    site_name = bp_patches.name
+    return dict(site_name=site_name,)
+
+
+# 403 and 404 errors
+bp_patches.errorhandler(403)(
+    lambda e: (ErrorHandler()(e, http=403), 403))
+bp_patches.errorhandler(404)(
+    lambda e: (ErrorHandler()(e, http=404), 404))
+
+
+# INDEXVIEW
+bp_patches.add_url_rule(
+    '/',
+    view_func=IndexView.as_view(
+        'index',
+        render_func=bind_render('patches/index.html'),
+        err_func=ErrorHandler('patches'),
+        news_html='patches_news.html'))
+
+# ping service
+bp_patches.add_url_rule(
+    '/api/ping/',
+    view_func=Ping.as_view(
+        'ping',))
+
+# PREFIXVIEW
+bp_patches.add_url_rule(
+    '/prefix/<prefix>/',
+    view_func=PrefixView.as_view(
+        'prefix',
+        render_func=bind_render('prefix.html'),
+        err_func=ErrorHandler('patches'),))
+
+
+# api
+bp_patches.add_url_rule(
+    '/api/prefix/<prefix>/',
+    view_func=PrefixView.as_view(
+        'api_prefix',
+        render_func=jsonify,
+        err_func=ErrorHandler(mode='json')))
+
+
+# LISTPACKAGESVIEW
+bp_patches.add_url_rule(
+    '/list/<int:page>/',
+    view_func=ListPackagesView.as_view(
+        'list_packages',
+        render_func=bind_render('list.html'),
+        err_func=ErrorHandler('patches'),
+        pagination=True))
+
+
+# api
+bp_patches.add_url_rule(
+    '/api/list/',
+    view_func=ListPackagesView.as_view(
+        'api_list_packages',
+        render_func=jsonify,
+        err_func=ErrorHandler(mode='json')))
+
+# VERSIONSVIEW
+bp_patches.add_url_rule(
+    '/summary/<string:packagename>/',
+    view_func=PackageVersionsView.as_view(
+        'versions',
+        render_func=bind_render('patches/package.html'),
+        err_func=ErrorHandler('patches')))
+
+# api
+bp_patches.add_url_rule(
+    '/api/summary/<string:packagename>/',
+    view_func=PackageVersionsView.as_view(
+        'api_patch_versions',
+        render_func=jsonify,
+        err_func=ErrorHandler(mode='json')))
+
+# PATCHVIEW
+bp_patches.add_url_rule(
+    '/summary/<path:path_to>/',
+    view_func=SummaryView.as_view(
+        'summary',
+        render_func=bind_render('patches/summary.html'),
+        err_func=ErrorHandler('patches')))
```

Comment:
2 notes here:
* /summary/\<path> could be replaced by /summary/\<package>/\<version>, since that's the "deepest" this blueprint will go, right?
* why not /api/summary/\<path> ?